api/ipfsproxy/ipfsproxy.go

```diff
@@ -647,6 +651,50 @@ func (proxy *Server) repoStatHandler(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+type ipfsRepoGCResp struct {
+	Key   cid.Cid `json:",omitempty"`
+	Error string  `json:",omitempty"`
+}
+
+func (proxy *Server) repoGCHandler(w http.ResponseWriter, r *http.Request) {
+	queryValues := r.URL.Query()
+	streamErrors := queryValues.Get("stream-errors")
+	// TODO: Does `quiet` really quiet anything?
+
+	proxy.setHeaders(w.Header(), r)
+	var repoGC api.GlobalRepoGC
+	err := proxy.rpcClient.CallContext(
+		r.Context(),
+		"",
+		"Cluster",
+		"RepoGC",
+		struct{}{},
+		&repoGC,
+	)
+	if err != nil {
+		ipfsErrorResponder(w, err.Error(), -1)
+	}
+
+	w.WriteHeader(http.StatusOK)
+	enc := json.NewEncoder(w)
+	var ipfsRepoGC ipfsRepoGCResp
+	for _, gc := range repoGC.PeerMap {
+		for _, key := range gc.Keys {
+			if streamErrors == "true" {
+				ipfsRepoGC = ipfsRepoGCResp{Key: key.Key, Error: key.Error}
+			} else {
+				ipfsRepoGC = ipfsRepoGCResp{Key: key.Key}
+			}
+
+			// Cluster tags start with small letter, but IPFS tags with capital letter.
+			if err := enc.Encode(ipfsRepoGC); err != nil {
+				logger.Error(err)
+			}
+		}
+	}
+	return
```

Comment:
If stream errors is false and there were errors, these need to be collected and the X-Stream-Error Trailer needs to be set to their value (ipfs has a multierror type that is returned here (https://github.com/ipfs/go-ipfs/blob/master/core/commands/repo.go#L97). Similar to /add. The logic of how ipfs does this last part is around go-ipfs-cmds/http/responseEmitter:closeWithError().

Reply:
Worth testing.